src/common/regions/utils.ts

```diff
@@ -56,3 +60,77 @@ export function getAutocompleteRegions(region?: Region) {
     return [];
   }
 }
+
+function countyIncludesZip(county: County, zipCode: string): boolean {
+  return county.zipCodes.includes(zipCode);
+}
+
+export function getStateRegionFromStateCode(
+  stateCode: string,
+): Region | undefined {
+  const regionFromStateCode = find(
+    regions.states,
+    (region: State) => region.stateCode === stateCode,
+  );
+  return regionFromStateCode;
+}
+
+export function getCountyRegionFromZipCode(
+  zipCode: string,
+): Region | undefined {
+  const countyFromZip = find(regions.counties, (region: County) =>
+    countyIncludesZip(region, zipCode),
+  );
+  return countyFromZip;
+}
+
+// TODO (Chelsi): fix the any
+export function getMetroRegionFromZipCode(zipCode: string): any {
+  const metroFromZip = find(regions.metroAreas, (region: MetroArea) =>
+    find(region.counties, (region: County) =>
+      countyIncludesZip(region, zipCode),
+    ),
+  );
+  return metroFromZip;
+}
+
+// TODO (chelsi): move elsewhere?
+interface GeolocatedRegions {
+  county?: Region;
+  metroArea?: Region;
+  state?: Region;
+}
+
+export function getGeolocatedRegions(
+  geolocation: GeolocationInfo,
+): GeolocatedRegions | null {
+  if (geolocation.country !== UNITED_STATES) {
+    return null;
+  } else {
+    return {
+      county: getCountyRegionFromZipCode(geolocation.zipCode),
+      metroArea: getMetroRegionFromZipCode(geolocation.zipCode),
+      state: getStateRegionFromStateCode(geolocation.stateCode),
+    };
+  }
+}
+
+/**
+ * If we are able to geolocate the user, we rank their regions (state, county, metro if applicable) first
+ * in the searchbar dropdown menu. This sorts the Region[] accordingly.
+ */
+export function getAutocompleteRegionsWithGeolocation(
+  geolocation: GeolocationInfo,
+  locations: Region[],
+): Region[] {
+  const geolocatedRegions = getGeolocatedRegions(geolocation);
```

Comment:
Returning an array of `Region` object (with 0, 1, 2, or 3 items) will eliminate the need of sanitizing